app/models/podcast.rb

```diff
@@ -15,7 +15,9 @@ class Podcast < ApplicationRecord
 
   after_save :bust_cache
 
-  scope :reachable, -> { where(reachable: true) }
+  scope :reachable, lambda {
```

Comment:
I think this can be easier in its definition:

```ruby
scope :reachable, -> { joins(:podcast_episodes).where(podcast_episodes: {reachable: true}) }
```

Reply:
It looks simpler, but `joins` + `where` is slower + there is a need to add `distinct`.

Reply:
You're absolutely right, what about this version?

```ruby
Podcast.where(id: PodcastEpisode.where(reachable: true).select(:podcast_id))
```

which scoped would become:

```ruby
scope :reachable, -> { where(id: PodcastEpisode.where(reachable: true).select(:podcast_id)) }
```

The query plan and the result are the same:

```sql
PracticalDeveloper_development=# explain analyze SELECT "podcasts".* FROM "podcasts" WHERE (EXISTS(SELECT 1 from podcast_episodes WHERE podcast_episodes.podcast_id = podcasts.id AND podcast_episodes.reachable = true));
                                                       QUERY PLAN
-------------------------------------------------------------------------------------------------------------------------
 Hash Semi Join  (cost=12.11..13.21 rows=4 width=470) (actual time=0.048..0.054 rows=4 loops=1)
   Hash Cond: (podcasts.id = podcast_episodes.podcast_id)
   ->  Seq Scan on podcasts  (cost=0.00..1.04 rows=4 width=470) (actual time=0.016..0.019 rows=4 loops=1)
   ->  Hash  (cost=11.30..11.30 rows=65 width=4) (actual time=0.022..0.022 rows=12 loops=1)
         Buckets: 1024  Batches: 1  Memory Usage: 9kB
         ->  Seq Scan on podcast_episodes  (cost=0.00..11.30 rows=65 width=4) (actual time=0.008..0.014 rows=12 loops=1)
               Filter: reachable
 Planning Time: 0.239 ms
 Execution Time: 0.096 ms
(9 rows)

Time: 0.750 ms
PracticalDeveloper_development=# explain analyze SELECT "podcasts".* FROM "podcasts" WHERE "podcasts"."id" IN (SELECT "podcast_episodes"."podcast_id" FROM "podcast_episodes" WHERE "podcast_episodes"."reachable" = TRUE);
                                                       QUERY PLAN
-------------------------------------------------------------------------------------------------------------------------
 Hash Semi Join  (cost=12.11..13.21 rows=4 width=470) (actual time=0.047..0.053 rows=4 loops=1)
   Hash Cond: (podcasts.id = podcast_episodes.podcast_id)
   ->  Seq Scan on podcasts  (cost=0.00..1.04 rows=4 width=470) (actual time=0.014..0.018 rows=4 loops=1)
   ->  Hash  (cost=11.30..11.30 rows=65 width=4) (actual time=0.018..0.018 rows=12 loops=1)
         Buckets: 1024  Batches: 1  Memory Usage: 9kB
         ->  Seq Scan on podcast_episodes  (cost=0.00..11.30 rows=65 width=4) (actual time=0.007..0.012 rows=12 loops=1)
               Filter: reachable
 Planning Time: 0.196 ms
 Execution Time: 0.085 ms
(9 rows)
```

Reply:
which can even be simplified to

```ruby
scope :reachable, -> { where(id: PodcastEpisode.reachable.select(:podcast_id)) }
```

Reply:
I've benchmarked and this version seems even better :+1: